index.d.ts

```diff
@@ -22,11 +22,12 @@ declare interface GCStoreOptions extends DataStoreOptions {
 }
 
 declare interface S3StoreOptions extends DataStoreOptions {
-    accessKeyId: string;
-    secretAccessKey: string;
+    accessKeyId?: string;
+    secretAccessKey?: string;
+    credentials?: Object;
```

Comment:
This is not strict enough, we can now omit  `accessKeyId`, `secretAccessKey`, _and_ `credentials`.

1. Change types to allow only `credentials` _or_ `accessKeyId` and `secretAccessKey`.
   1. Here is an [TS playground example](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wBYxmOAzKCEOAIixxIG4AoUSWOAbzgFEA3AUwDsYWRhsHKHAC++QsRIdOPVFWrUAJhwDGAG2RQOcYDyF5kK7QBFkMZAGUY0DgHkwMYBC7pG1OB7hgzGAFxxUGChdAHMaTzguZH4uEIAxAFcuFUdnAH5-AAotAEd-cjAAOgBJZKJQgFkOVFRkEI4ASjgAXgA+AKDQ8M8tDUdOABkIFTMnLgy4ACMICDUOZC4aEQVldU1tXRh9Q20LAGYrGwAhNDsHMfQOAA8trkV0U3NDrXtUlyZ3T0mElQBrDhg-kCwVi3Q8WhCYwmwK6nw8MHAxmAUAAClo8MArtDOqC4V5NDALMAAF4cfxcBIgSZCJZKVQaLQ6PRQAxGOD7Z5nN6oADqAgwAGkOABPACCdwsqi08DcEUMRhqQuFxUUQJxYTxqClANFKgVqCVapBGoiKi0yh4wGQalQEy40hp1GWqwZG2ZrN2B2sL3Ozl5-IAwubuI5ra48fLqgaRSq7Q6oGCAtqYLr9Uq45wE3izRwLaGbf5bJMAFaqGCUOAAekrcBAyH+cBgGGA6AW7L2jeFYG0ToUMC7ns5rwuLTgmQ53q5Fz5TaDuZDVptcAAZO3OSctcO-U0AD5jic2LcuGeCkXixSSnPwVcHrQbqfbhTXejwF3rOBrGpwR6WSdwa63PcrAyLw-CCMIsqeCofpBD83qZBAvouP4P5DkhqANIm9TwAB3CoBcmQNP4CzComWo4TceEEbhLgXAA+poUDIMKxFcMKmHZloZgcNkHB5HAKIEouAA8BQlGUMQhFUNR1BwrREQJEgthwwkka0iYAO7BFsmR4hEuT5NgRSlNBknSbU9QADR6Z4GJzHRwCKNixrWREEQQHgeDkXalLUlAeIKWi5RaqpbHqXi2G2J55GZHZHAOaqHTGoFSkhWptLUM+0CvvS76fugt7aDRQGoX+kEeNBLiwSk0AIeh-iFUeGFYQCUVeQCsXAPZjlGqEllwBp-J0d4sC2v40yzPMXApcFKnpb21DVh+0BaCk1D2hpa6TpkzDfH8AL+AA5Id-UjYSJJknAACMp0+EdJ0fsGlphv4jAiCIDTrRwm2FTtUw-P8gJwMdt2wESpL+Dd+JNvd-WRoqMaJSDSZXqmUaGsDh0fQoS1ac4IRfT9Xo2H9e2A7D+JgxdkO3TDmPYxtW0k7tAMHZjoPnRD120347OPfOz0FkwIhw3q6OIxTWqo2LCMsfTn1LVwEDwBiVy5nAwoAoTTNaKTrNA8jZ3g5dUPeHTyM5nmi6oP4STKBi9qKKLaYS3zUvSmjsv3Qz306zxLP7QbD1G9T3PQ7zFtPfmNvC874vKoldscA7ub9e7Ooy9GctJynigfUAA) that is almost perfect. Feel free to improve it to catch the missing cases. 
2. `import type` the actual credentials type from the aws lib.

Reply:
If we don't export the `S3StoreOptions` type, then it's probably better to do method overloading ([example](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wBYxmOAzKCEOAIixxIG4AoUSWOAbzgFEA3AUwDsYWRhsHKHAC++QsRIdOPVFWrUOAD3rxgPIXmQBjDnAAiyGMgDKMaBwDyYGMAhd0janBdwwRjAC44qGFHUA5jSucFzI-FwBAGIArlzatvYA-N4AFFAcAI7e5GAAdACS8USBALIcqKjIARwAlHAAvAB8Pn6Bwa4ZADZGwJwAMhDavcneAEYQEF0cyFw0IgrKqnDqMJo6eiYAzGYW1okOcMprXAAm6IbGuxn7dodOIWMx2gDWHDDevv6RHS4ZAXcUq1vkFnK4YOB9MAoAAFDJ4YBKIFfdpglzuWAmYAALw43i4MRAYyE80WKmg8FOHG0PQycBpaAuRlM5jpx2451YMl4-EEwgerm09i+z1ZqQgNjuqG8lxZe0lwtqvzgNXg7IcUtStW8swAnsrUO8jkoTqhNeqzcKAPrIKBQZC6nVcXVKtH0jJGDjpLLeGG22zILoAHlyhWKEQC5Uq1Q4TW1cDhJUNQb1TWVAHd-GtUm6QhlsphsPkikKI1GqjUADS51wI6ZW4CnZFtSLVkIhCB4PCGmBAglEoRu+OJ-jJ1PK1WWLs91J1jgN06fFsBYcSYBj51p6gLRTkhhUhl0hmVODba56dWc2XnphuoUOPyi6DihUObxn1lWV-oABkTBQ2i6JUADSHC6gUi7AoElY+NSGQwAAgoBFSoKBjpQZEoiuiE94igkz4Sgc0qnjsn63MKcB-sw2gZFSPDAIGxEQGMABW1LwCI2GuJO07vLOwD1o2S4gjB6YCBgVoYjAqBAhMUwzFwq5JhwKabqS1AAPQafS0AZAk1BcBw6YkeeqTME8rzvN4ADk1kwVJWK4t4ACM9keDZdnuhwdEBl0xGMCInEGUZJmfmZcAWW8HxwLZbmYjieJwK5bjuTFnk6EBqFgRBHkwYaNHvEhmVoR5QWadpmb2AEwXGR+FjhZFVlpXFMCOYlyXuDAXhpUFhm1aR9Xmc8UW5Sl8VOUlbldaNBU+QxfneAFMEZShaE5c1sEFYhyEgWBpW1DVoWDRFw1NbFY2tQlLlTd152zdwvnEXEVIIoZpzLTtWXgZBz0cK93l5XBhWfSVcC-f9py9SFdUZA1p3RedDlXZNKXTRt930Yxi0iB9xXZZB535fBRWrXtYNnH96jeVD-WmUNlkI55SMTR1qV3bRD3zU9FMQ5xQA)) 🤔